mesonbuild/mesonlib.py

```diff
@@ -22,6 +22,15 @@
 
 from glob import glob
 
+if sys.executable.endswith('meson.exe'):
```

Comment:
Here and in other places, `endswith` seems fragile. What about `os.path.basename(sys.executable) == 'meson.exe'` instead?